finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
```

Comment:
Here is an [alternate implementation](https://github.com/koshelev/finagle/commit/b86a94c669ef91148283435be00e95e13b48811a), with a handler deferring the `channelRegistered` event. As we still need ChannelTransport, it will be added in `handlerAdded`. As consequence `ServerBridge` is not sharable any more. What do you think?